internal/pkg/merge/merge.go

```diff
@@ -0,0 +1,106 @@
+// ------------------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 bomctl authors
+// SPDX-FileName: internal/pkg/export/export.go
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// ------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+package merge
+
+import (
+	"fmt"
+	"os"
+	"slices"
+
+	"github.com/charmbracelet/log"
+	"github.com/google/uuid"
+	"github.com/protobom/protobom/pkg/sbom"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/bomctl/bomctl/internal/pkg/db"
+)
+
+type (
+	MergeOptions struct {
+		Logger       *log.Logger
+		OutputFile   *os.File
+		CacheDir     string
+		ConfigFile   string
+		Debug        bool
+		Expand		 bool
+		Name         string
+	}
+)
```

Comment:
You'll need to fetch the upstream and resolve conflicts, but these go into `options.go` now, as something like

```go
	MergeOptions struct {
		OutputFile *os.File
		Expand     bool
		Name       string
	}
```